hal/src/photon/softap.cpp

```diff
@@ -1397,7 +1397,17 @@ class HTTPDispatcher {
             } else {
                 PageProvider* p = (PageProvider*)arg;
                 if (p) {
-                    p(req->url ? req->url : url, &writeHeader, req->stream, &r, &w, nullptr);
+                    char* baseurl = (char*)(req->url ? req->url : url);
+                    const size_t baseurl_len = strlen(baseurl);
```

Comment:
`strlen(baseurl)` is called before the `baseurl != nullptr` check below. I'm not sure if WICED can provide an application with a null URL string, but let's make the code consistent